File: node/PostOffice/src/models/Package.ts
import {DeliveryType} from "./DeliveryType";

export class Package {
  private id: string;
  private height: number;
  private width: number;
  private length: number;
  private weight: number;
  private deliveryType: DeliveryType;

  constructor(id: string, height: number, width: number, length: number, weight: number, deliveryType: DeliveryType) {
    this.id = id;
    this.height = height;
    this.width = width;
    this.length = length;
    this.weight = weight;
    this.deliveryType = deliveryType;
  }

  // Getters
  public getId(): string {
    return this.id;
  }

  public getHeight(): number {
    return this.height;
  }

  public getWidth(): number {
    return this.width;
  }

  public getLength(): number {
    return this.length;
  }

  public getWeight(): number {
    return this.weight;
  }

  public getDeliveryType(): DeliveryType {
    return this.deliveryType;
  }

  // Setters
  public setId(id: string): void {
    this.id = id;
  }

  public setHeight(height: number): void {
    this.height = height;
  }

  public setWidth(width: number): void {
    this.width = width;
  }

  public setLength(length: number): void {
    this.length = length;
  }

  public setWeight(weight: number): void {
    this.weight = weight;
  }

  public setDeliveryType(deliveryType: DeliveryType): void {
    this.deliveryType = deliveryType;
  }

  public toString(): string {
    return `Package{id=${this.id}, height=${this.height}, width=${this.width}, length=${this.length}, weight=${this.weight}, deliveryType=${this.deliveryType}}`;
  }
}

File: java/cucumber-archetype/src/main/java/postoffice/models/Package.java
package postoffice.models;

import java.util.UUID;

public class Package {
    private UUID id;
    private int height;
    private int width;
    private int length;
    private double weight;
    private DeliveryType deliveryType;

    public Package() {

    }

    // Constructor
    public Package(UUID id, int height, int width, int length, double weight, DeliveryType deliveryType) {
        this.id = id;
        this.height = height;
        this.width = width;
        this.length = length;
        this.weight = weight;
        this.deliveryType = deliveryType;
    }

    // Getters
    public UUID getId() {
        return id;
    }

    public int getHeight() {
        return height;
    }

    public int getWidth() {
        return width;
    }

    public int getLength() {
        return length;
    }

    public double getWeight() {
        return weight;
    }

    // Setters
    public void setId(UUID id) {
        this.id = id;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public void setLength(int length) {
        this.length = length;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    // Getters and Setters for DeliveryType
    public DeliveryType getDeliveryType() {
        return deliveryType;
    }

    public void setDeliveryType(DeliveryType deliveryType) {
        this.deliveryType = deliveryType;
    }

    @Override
    public String toString() {
        return "Package{" +
                "id=" + id +
                ", height=" + height +
                ", width=" + width +
                ", length=" + length +
                ", weight=" + weight +
                ", deliveryType=" + deliveryType +
                '}';
    }
}